Newsletter Settings: add clarify to clarify reply-to settings

jeherve opened the pull request:
Fixes CML-934

## Proposed changes:

When one enables the option for replies to be sent to the post author's email, this will only work if the post author has linked their account to their WordPress.com account.

We want to highlight that in the newsletter settings. We've discussed multiple options in CML-934, and ended up with this choice:

<img width="1091" height="366" alt="image" src="https://github.com/user-attachments/assets/d6131de7-8dd0-4982-ad04-c6b40cbe10c9" />

<img width="1059" height="365" alt="image" src="https://github.com/user-attachments/assets/39a423fc-087b-4724-b807-99ca50f65c26" />

> show the highlighted text when the “Replies will be sent to the post author’s email” option is selected AND the current site is self-hosted?

Since we only want to show that option on self-hosted sites, we won't be adding it to the Calypso UI here:
https://github.com/Automattic/wp-calypso/blob/a6947294442abc306659ccf8c4e2e35fd9460edd/client/my-sites/site-settings/settings-newsletter/ReplyToSetting.tsx#L53

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* See CML-934

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Go to Jetpack > Settings > Newsletter
* Enable the feature
* Scroll down to "Sender name and reply-to settings"
* Pick "Replies will be sent to the post author's email"
* See the text changing just above.

jeherve committed:
Newsletter Settings: add clarify to clarify reply-to settings

Fixes CML-934

Addison-Stavlo reviewed (dismissed):
This works as expected for me!

Aside, my 2cents: it feels weird having this new descriptor conditionally in the headline above the settings. Being detached from the specific relevant setting, having a delay after clicking the setting, the message showing up appended to another line further above after the delay.... etc.... it feels easier to miss compared to just putting on the individual option's description directly.

jeherve committed:
Separate sender and reply-to into 2 sections

And move the explanations below the radio / fields.

This will bring this settings section more in line with its Calypso counterpart.

jeherve commented:
> Aside, my 2cents: it feels weird having this new descriptor conditionally in the headline above the settings.

That's a good point. The Calypso settings actually look a bit better in that regard, if only because the explanations are displayed below the fields and because we have 2 separate sections. I think we could improve things a bit by making the settings here more like the Calypso ones. 

I just pushed a commit, ce8fafedea2643adc4a8ed3cdcec7223f95d0b55, to give you an idea of what I think:

**Before**

<img width="971" height="887" alt="Screenshot 2025-10-03 at 19 00 42" src="https://github.com/user-attachments/assets/f1ca8028-1388-46ea-b017-3169796fb3f7" />

**After**

<img width="937" height="891" alt="image" src="https://github.com/user-attachments/assets/e02a70a7-3e10-42bf-bc43-ce27bb519339" />

Let me know what you think. It doesn't address all your concerns, but I think it makes the extra appended message a bit more bearable.
